src/OpenTelemetry.Resources.AssemblyMetadata/README.md

```diff
@@ -0,0 +1,137 @@
+## Usage
+
+You can add the Assembly Metadata Resource Detector to the `ResourceBuilder` with the following example:
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Resources;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddOpenTelemetry()
+    .ConfigureResource(static builder => builder.AddAssemblyMetadataDetector())
+    .WithLogging(static builder => /* ... */)
+    .WithMetrics(static builder => /* ... */)
+    .WithTracing(static builder => /* ... */);
+```
+
+The resource detector will record attributes defined on the [entry assembly](https://learn.microsoft.com/dotnet/api/system.reflection.assembly.getentryassembly) using [AssemblyMetadataAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblymetadataattribute) with keys prefixed by `otel:`.
+
+The `OpenTelemetry.Resources.AssemblyMetadata` package includes an MSBuild `.targets` file which defines a new item type, `OtelResourceAttribute`, which can be used to define resource attributes in MSBuild project files.
+
+The following example demonstrates how you can use pre-existing [SourceLink](https://github.com/dotnet/sourcelink) metadata
+to populate the [`vcs.ref.head.name`](https://opentelemetry.io/docs/specs/semconv/registry/attributes/vcs/#vcs-ref-head-name), [`vcs.ref.head.revision`](https://opentelemetry.io/docs/specs/semconv/registry/attributes/vcs/#vcs-ref-head-revision), and [`vcs.repository.url.full`](https://opentelemetry.io/docs/specs/semconv/registry/attributes/vcs/#vcs-repository-url-full) attributes:
+
+```csproj
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>...</PropertyGroup>
+
+  <ItemGroup>
+    <OtelResourceAttribute Include="vcs.ref.head.name" Value="$(SourceBranchName)" />
+    <OtelResourceAttribute Include="vcs.ref.head.revision" Value="$(SourceRevisionId)" />
+    <OtelResourceAttribute Include="vcs.repository.url.full" Value="$(RepositoryUrl)" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry.Resources.AssemblyMetadata" />
+  </ItemGroup>
+
+</Project>
+```
+
+Alternatively, these attributes could be set from user-defined environment variables set in the build environment:
+
+```csproj
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>...</PropertyGroup>
+
+  <ItemGroup>
+    <OtelResourceAttribute Include="vcs.ref.head.name" Value="$(VCS_REF_HEAD_NAME)" />
+    <OtelResourceAttribute Include="vcs.ref.head.revision" Value="$(VCS_REF_HEAD_REVISION)" />
+    <OtelResourceAttribute Include="vcs.repository.url.full" Value="$(VCS_REPOSITORY_URL_FULL)" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry.Resources.AssemblyMetadata" />
+  </ItemGroup>
+
+</Project>
+```
+
+You can check on the output from these items by inspecting the `$(AssemblyName).AssemblyInfo.cs` file in your intermediate output path (e.g. `SomeProject/obj/debug/SomeProject.AssemblyInfo.cs`):
```

Comment:
It feels too detailed to include this generated code example in the README, IMHO. For me, this is more of a compiler implementation detail than one for the resource detector.